go/cmd/dolt/commands/sqlserver/serverconfig.go

```diff
@@ -171,11 +171,23 @@ func (cfg *commandLineServerConfig) Password() string {
 
 // ReadTimeout returns the read and write timeouts.
 func (cfg *commandLineServerConfig) ReadTimeout() uint64 {
+	if cfg.timeout == 0 && cfg.PersistenceBehavior() == loadPerisistentGlobals {
```

Comment:
this does two things -- 1) applies a partial default hierarchy for config values, 2) makes the config dynamic to session level variable changes. I think 1) is somewhat confusing, because it fails to take into account command line arguments in the hierarchy. 2) is an enhancement, but these configs are only ever used once immediately at Serve time, so it is not used. Perhaps it is better to apply the full hierarchy elsewhere?

Reply:
I included getting persisted global values in `getCommandLineServerConfig` but it's kind of weird. I was not sure how to set timeout when there are two types of timeouts. At the same time, the timeout are set through `ReadTimeout()` and `WriteTimeout()` which both return the same value, `timeout`.